src/org/mozilla/javascript/NativeJavaObject.java

```diff
@@ -88,6 +85,9 @@ public boolean has(Symbol key, Scriptable start) {
 
     @Override
     public Object get(String name, Scriptable start) {
+        if (name.equals(NativeIterator.ITERATOR_PROPERTY_NAME) && javaObject instanceof Iterable) {
+            return es5Iterator;
```

Comment:
There is no iterator in the ES5 specification.
Rhino is a JavaScript engine. Please check the JavaScript specification in [ECMA262](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/) or [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript) before making any changes.

Reply:
Thanks for the hint and I ask for your indulgence when I try to submit one or the other nonsensical change ;)

I have to admit that I don't know the specification in detail and have only even rudimentarily read it so far. 
In particular, I have trouble telling the difference between `value[Symbol.iterator]` and `value["__iterator__"]`. Especially finding the correct documentation

As far I understand now, the `__iterator__`  is a very old thing (https://web.archive.org/web/20061203042508/http://developer.mozilla.org:80/en/docs/New_in_JavaScript_1.7%23Iterators) and is used when looping with `for...in` or `for each ... in` (for each in is also deprecated/not recommended to use, as I remember from other comments)

The `Symbol.iterator` was introduced with ES5 and works completely different than the above one. It is used in `for..of` loops.

> Rhino is a JavaScript engine.
... and here we are trying to use java objects in javascript (which isn't specified yet, AFAIK)

So, what this PR enables is to iterate over any `java.util.Iterable` with `for...in` / `for each ... in`.

and now, while writing these lines, the penny dropped...
1. It doesn't make much sense to use `for...in` with a `java.util.HashSet` as you might get only the IDs, but you cannot access them by `hashSet[id]`
2. The `for each ... in` is deprecated and Rhino specific. It can be easily replaced by `for ... of`

So the advantage of this PR is, that you can use the (deprecated) `for each` on iterable objects.

Unfortunately, we have used "for each" in our code and also the patch in this PR, so the clean way is to migrate our js code to `for...of`.

Thanks again
Roland